### ISSUE TITLE
Magento should catch getimagesize() issues

### ISSUE THREAD

**Issue by loekvangool** [OpenMage/magento-lts#943]
If `getimagesize()` runs into trouble, a `NOTICE` is thrown and code execution continues, no biggie. But, Magento assumes the operation went fine. But we all know that images can go missing or get corrupted.

For instance in https://github.com/OpenMage/magento-lts/blob/1.9.4.x/app/code/core/Mage/Catalog/Model/Product/Image.php#L221. Accessing `$imageInfo[0]` will then subsequently lead to a `FATAL` over a corrupt or non-existing image.

There are 13 more usages of `getimagesize()` at this time: https://github.com/OpenMage/magento-lts/search?q=getimagesize&unscoped_q=getimagesize.

Magento should handle these situations more gracefully IMO.

**Comment by colinmollenhour** [comment]
Agreed, ideally each instance should be checking the return value to ensure it is not `FALSE` and then throwing an error early if so, like "Could not get image size for provided file.". In developer mode these errors are thrown as exceptions so it would only have an effect in production mode.
